feat(snap): add snap install method to `install.sh` when snap is available (#1697)
NVIDIA/OpenShell: docs/about/installation.mdx

## Linux

On distributions that ship with `snapd`, the install script uses the Snap path

> described below, provided no non-snap Docker is detected. If a native Docker
> package is present, the installer silently falls back to DEB or RPM. On hosts
> without `snapd` (or with
> `OPENSHELL_INSTALL_METHOD=classic` set), the script falls back to the classic

Reviewer: ```suggestion
described below, unless one of the following is true:
- a native (non-snap) Docker package is present (the `openshell` snap requires the `docker` snap, for now)
- `snapd` is not present
- `OPENSHELL_INSTALL_METHOD=classic` is set in the environment
```

## Snap

On Linux distributions that ship with `snapd`, the install script installs
OpenShell from the Snap Store. The OpenShell snap bundles the CLI, the terminal

> UI, and a managed gateway daemon. snapd handles upgrades and rollback; the

Reviewer: ```suggestion
UI, and a managed gateway daemon. `snapd` handles upgrades and rollback; the
```

gateway runs as a system service inside the snap.

You can also install the snap directly:

```shell

> sudo snap install openshell

Reviewer: Since the `docker` snap is currently required, should we include this here as well?

```suggestion
sudo snap install docker
sudo snap install openshell
```

`/var/snap/openshell/common/gateway.db`). Create `$SNAP_COMMON/gateway.toml`
when you need to override gateway settings.

> The snap requires the Docker snap. `default-provider: docker` on the OpenShell
> snap installs the Docker snap automatically on first use, but you can install
> it up front with:

Reviewer: ```suggestion
The OpenShell snap requires the Docker snap, which you can install with:
```

```shell
sudo snap install docker
```

### Snap store installs

> When installing from the Snap Store, snapd automatically connects the `home`
> and `network` plugs. The remaining plugs — `network-bind`, `ssh-keys`,
> `log-observe`, `system-observe`, and `docker` — still require manual connection:

Reviewer: I believe `network-bind` auto-connects as well

```suggestion
When installing from the Snap Store, snapd automatically connects the `home`,
`network`, and `network-bind` plugs. The remaining plugs — `docker`,
`log-observe`, `ssh-keys`, and `system-observe` — still require manual
connection:
```

```shell
sudo snap connect openshell:docker docker:docker-daemon
sudo snap connect openshell:log-observe
sudo snap connect openshell:ssh-keys
sudo snap connect openshell:system-observe
```

> The Docker slot is the Docker snap's `docker-daemon` slot; OpenShell does not
> work with a host-installed Docker Engine. The snap declares `default-provider:
> docker` on the Docker plug so snapd will offer to install the Docker snap, but
> the connection itself must be made manually. The installer runs these
> connections for you on Snap installs; run them by hand if you install the snap
> manually. The installer is best-effort: if a connect fails (for example because
> the Docker snap is not yet running), the snap still installs and the installer
> prints a warning.

Reviewer: There's no `default-provider` anymore, and I think the wording can be simplified a bit.

```suggestion
The `docker` interface is used to connect the `openshell` snap's `docker` plug
to the `docker` snap's `docker-daemon` slot; the `openshell` snap does not yet
work with a host-installed Docker Engine. The installer runs this and the other
interface connections for you after installing the `openshell` snap; run them
by hand if you install the snap manually. The installer is best-effort: if a
connect fails (for example because the `docker` snap is not yet running), the
snap still installs and the installer prints a warning.
```

sudo systemctl restart snap.openshell.gateway
```

### When to choose Snap

> Use Snap when `snapd` is available and no native Docker Engine is installed,
> and you want atomic upgrades and rollback, a single self-contained install that
> bundles the Docker provider, or a desktop launcher that surfaces the OpenShell
> terminal UI in the application menu.

Reviewer: `docker` is not specified as a `default-provider` anymore, so we should remove that language here.

```suggestion
Use the `openshell` snap when `snapd` is available and no native Docker Engine
is installed, and you want atomic upgrades and rollback, a single self-contained
and sandboxed install, or a desktop launcher that surfaces the OpenShell
terminal UI in the application menu.
```

> Use DEB or RPM when `snapd` is unavailable or when you already run Docker Engine
> from a non-snap source. The installer falls back to DEB or RPM on hosts with
> native Docker.

Reviewer: ```suggestion
Install via the `.deb` or `.rpm` when `snapd` is unavailable or when you already
run Docker from a non-snap source. The installer falls back to these methods on
hosts with native Docker.
```

Set `OPENSHELL_INSTALL_METHOD=classic` to force the classic package on hosts

> that also have `snapd` available.

Reviewer 2: instead of `OPENSHELL_INSTALL_METHOD=classic`, how about we enumerate all the valid installation methods?

Eg `OPENSHELL_INSTALL_METHOD=apt|homebrew|rpm|snap`
